ExtraDry/ExtraDry.Swashbuckle/Filters/ResourceReferenceSchemaFilter.cs

```diff
@@ -15,12 +15,13 @@ public class ResourceReferenceSchemaFilter : IDocumentFilter {
     public void Apply(OpenApiDocument swaggerDoc, DocumentFilterContext context)
     {
         LoadResourceReferences();
-        RewriteResourceReferenceTypes(swaggerDoc);
+        RewriteResourceReferenceTypes(swaggerDoc, context);
     }
 
-    private void RewriteResourceReferenceTypes(OpenApiDocument swaggerDoc)
+    private void RewriteResourceReferenceTypes(OpenApiDocument swaggerDoc, DocumentFilterContext context)
     {
-        foreach(var schema in swaggerDoc.Components.Schemas) {
+        for(int i = 0; i < swaggerDoc.Components.Schemas.Count; i++) {
```

Comment:
What is this achieving?  Is this a guard against null?  I.e. `swaggerDoc.Components.Schemas.Where(e => e != null)`?

Reply:
On line 35 we're generating a new schema which is added to the `Schemas` collection as part of the `GenerateSchema` call.  This would cause the `foreach` to error with an `InvalidOperationException` due to the collection being modified.

Reply:
I've added a comment to provide some clarity to what's happening in this section.

Reply:
@aakison  I've updated this based on your feedback.